mediaplayerbackend.cpp

```diff
@@ -372,13 +386,50 @@ void MediaPlayerBackend::onTracksReceived(const mopidy::Tracks &tracks)
 
     // TODO: from and until count
     for (auto track : tracks) {
+#ifndef QTIVI_NO_TAGLIB
+        if (!track.uri.endsWith(QLatin1String("mp3")))
+            break;
+
+        // Extract cover art
```

Comment:
Consider improve the comment to explain what happens, it's not that obvious that if there is no png-file with the same name as the mp3 file, it will check if the mp3 file contains an embedded icon, and then extracted the image and write it to a new, created png-file, next to the mp3-file.

Actually, I'm not really sure if this is what an end-user expects, i.e. that the system / media player modifies the file system.

I would prefer that for now, we don't support embedded images if there is a problem to provide the album art as an image to the UI.

Reply:
I will for sure add the comment, but I disagree with holding on this feature. This is an essential feature that makes a lot of visual difference.

We could for sure improve the way PNG files are stored, but this is how the simulation plugin works and nobody has ever complained.

Reply:
I dont' mean to skip the complete album art feature, it's needed. What I mean is that embedded images are converted into new png-files next to the music-file. I would recommend to only support existent png/jpg-files next to the music-file.

> We could for sure improve the way PNG files are stored, but this is how the simulation plugin works and nobody has ever complained.

For a simulation, that might be ok. So you mean that this is how it's actually implemented in the qt ivi media simulation backend?

Reply:
> I would recommend to only support existent png/jpg-files next to the music-file.

Why? It takes nothing to extract them.

> So you mean that this is how it's actually implemented in the qt ivi media simulation backend?

Yes, this code is virtually a copy-paste from the simulation plugin.

Reply:
> Why? It takes nothing to extract them.
Since the current solution modifies the FS. The alternative would be to create the file on the tmpfs.

Reply:
After discussion, let's keep the implementation as is since this is how the simulation backend behaves. Please create a story in the backlog to improve it.